### PR TITLE
Dgl graph store

### PR DESCRIPTION
**Pull request by BradReesWork**
new GraphStore for DGL intergration
Code will need to be added to DGL

### CODE REVIEW

File: python/cugraph/cugraph/gnn/graph_store.py [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+    @property
+    def gdata(self):
+        return self.__G
+
+    def __init__(self, graph=None, cluster=None):
+        if graph is not None:
+            if isinstance(graph, PropertyGraph):
+                self.__G = graph
+            else:
+                raise ValueError("graph must be a PropertyGraph")
```
Comment (rlratzel) [inline python/cugraph/cugraph/gnn/graph_store.py:50]: It looks like this is the only place a caller can set the input graph, so maybe it shouldn't be optional?
```suggestion
    def __init__(self, graph, cluster=None):
        if isinstance(graph, PropertyGraph):
            self.__G = graph
        else:
            raise ValueError("graph must be a PropertyGraph")
```

File: python/cugraph/cugraph/gnn/graph_store.py [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+
+        self.__client = cluster
+
+    ######################################
+    # Utilities
+    ######################################
+    @property
+    def num_vertices(self):
+        if self.__G is None:
+            raise ValueError("graph has not been set")
```
Comment (rlratzel) [inline python/cugraph/cugraph/gnn/graph_store.py:60]: if the above suggestion for the ctor is done, this and all others should be able to be safely removed.

File: python/cugraph/cugraph/gnn/graph_store.py [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+    then adds functions to match the DGL GraphStorage API.
+    However, this class only return cuGraph types.  Conversion to other types,
+    like DGLGraph, needs to be done in DGL or other external packages.
+
+    homogeneous graphs, graphs with no attributes - use Property Graph
+    hetrogeneous graphs - use PropertyGraph
+    """
+
+    __G = None
+    __clinet = None
```
Comment (rlratzel) [inline python/cugraph/cugraph/gnn/graph_store.py:31]: I think best practice is to only use class variable for constants and set all the instance variables in `__init__`.

File: python/cugraph/cugraph/gnn/graph_store.py (hunk 2 above)
Comment (BradReesWork) [reply python/cugraph/cugraph/gnn/graph_store.py:60]: dropping checks

File: python/cugraph/cugraph/gnn/graph_store.py (hunk 3 above)
Comment (BradReesWork) [reply python/cugraph/cugraph/gnn/graph_store.py:31]: updated

**Review by rlratzel — APPROVED** [review]

**Comment by BradReesWork** [comment]
@gpucibot merge

**Comment by wangxiaoyunNV** [comment]
Hi Brad, 
When I review the graphStorage class, I noticed that here is no output_device parameter in the sampling functions. But the DGL graphstorage sampling APIs have output_device parameter. Andrei's code has that as well. Could wee add that parameter to the samplings functions? or have another function called to_device in the GraphStorage class? 
https://github.com/dmlc/dgl/blob/701b4fccc2eed979ae3db801fabb6bf7bc03940c/examples/pytorch/__temporary__/dglnew/graph/graph.py

Thank you,
Xiaoyun

**Review by wangxiaoyunNV — COMMENTED** [review]
I have a concern about CuGraphStore class, it should have a to_device function or parameter in the sampling functions. 

Thank you!

**Pull request merged**